typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -98,13 +106,28 @@ async function main(address: string, options: { output: string }): Promise<void>
   process.on("SIGINT", () => {
     log("shutting down...");
     controller.abort();
+    isProcessing = false;
   });
 
+  async function processMsgQueue() {
+    while (isProcessing) {
+      let message = messageQueue.shift();
+      while (message != undefined) {
+        await writer.addMessage(message);
+        message = messageQueue.shift();
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
Instead of a tight loop with a timeout, you could use `promise-queue`:

https://github.com/foxglove/mcap/blob/83c0428e93ae88df6f7360d9461d569be966fd0b/website/src/components/McapRecordingDemo/Recorder.ts#L47-L48

Reply:
thanks for the suggestion, just implemented it